evergreen/run-tests.sh

```diff
@@ -102,26 +102,45 @@ if [[ "$OS" =~ Windows|windows ]]; then
     certutil.exe -urlcache localhost delete # clear the OS-level cache of all entries with the URL "localhost"
   fi
 else
-  export TARGET="Test"
+  if [[ "$FRAMEWORK" == "nil" ]]; then
+    export TARGET="Test"
+  else
+    export TARGET="Test${FRAMEWORK}"
+  fi
 fi
 
 echo "Final MongoDB_URI: $MONGODB_URI"
 if [ "$TOPOLOGY" == "sharded_cluster" ]; then
   echo "Final MongoDB URI with multiple mongoses: $MONGODB_URI_WITH_MULTIPLE_MONGOSES"
 fi
 for var in TMP TEMP NUGET_PACKAGES NUGET_HTTP_CACHE_PATH APPDATA; do
-  export $var=z:\\data\\tmp;
+  if [ "$OSTYPE" = "cygwin" ]; then
+    export $var=z:\\data\\tmp;
+  else
+    export $var=/data/tmp;
+  fi
```

Comment:
This pattern repeats often. We should define a global TMP_PATH variable that we set initially based on the OS and then use $TMP_PATH as needed.

Reply:
Sometimes we use `if [ COND ]` and other times `if [[ COND ]]`. The latter is more modern and preferred as it provides additional features. We should standardize on the latter throughout our Bash scripts.

Reply:
Can we defer this change? It seems stylistic.